sdk/nodejs/dynamic/index.ts

```diff
@@ -161,8 +161,22 @@ export interface ResourceProvider {
     delete?: (id: resource.ID, props: any) => Promise<void>;
 }
 
+const providerCache = new Map<ResourceProvider, Promise<string>>();
```

Comment:
Total nit - as I don't think it matters at all in practice - but this may be a case where [`WeakMap`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakMap) is actually appropriate.

Reply:
Ah, TIL. Will update.